pep-0394.txt

```diff
@@ -62,14 +66,35 @@ Recommendation
   context.
 * One exception to this is scripts that are deliberately written to be source
   compatible with both Python 2.x and 3.x. Such scripts may continue to use
-  ``python`` on their shebang line without affecting their portability.
+  ``python`` on their shebang line.
+* When packaging software that is source compatible with both versions,
+  distributions may change such ``python`` shebangs to ``python3`` (or
+  ``python2``). This ensures software is used with the latest version of
+  Python available, and it can remove a dependency on Python 2.
 * When reinvoking the interpreter from a Python script, querying
   ``sys.executable`` to avoid hardcoded assumptions regarding the
   interpreter location remains the preferred approach.
+* In controlled environments aimed at expert users, where being explicit
+  is valued over user experience (for example, in test environments and
+  package build systems), distributions may choose to not provide the
+  ``python`` command even if ``python2`` is available.
+  (All software in such a controlled environment must use ``python3`` or
+  ``python2`` rather than ``python``, which means scripts that deliberately
+  use ``python`` need to be modified for such environments.)
+* System administrators and users may be empowered to change the meaning of
+  the ``python`` command to ``python3``. This should be a deliberate action
+  by the administrator. Related documentation should include a discussion
+  of the effects -- for example, a link to the "Migration Notes" section of
+  this PEP.
+* When a virtual environment (created by the PEP 405 ``venv`` package or a
+  similar tool) is active, the ``python`` command should refer to the
+  virtual environment's interpreter. In other words, activating a virtual
+  environment counts as deliberate user action to change the default
+  ``python`` interpreter.
```

Comment:
+1 on this bullet point. Having `python` point to Python 3 when a Python 3 environment is activated is now established practice, and it would be nice for the PEP to acknowledge that. Many subparts of the community have already shifted away from the old idiom of relying on the system Python for user scripts.

Reply:
Yeah, I won't ask venv to change, I just `rm <env>/bin/python` after creating the env. :-) I wonder if it should say "may refer" rather than "should refer"?

I do note that this makes using `#!/usr/bin/env python` quite unreliable to invoke Python 2.

Reply:
I'd rather keep "should". It's quite an engrained expectation by now to have `python` point to the environment's Python. Doing otherwise would break those expectations.

Reply:
Aye, this is definitely a `should` - we rely on it heavily in the Python Packaging User Guide.

Reply:
OK.